File: FrontendCollaboration/WebContent/services/friendservice.js


app.factory('FriendService',function($http) {
	var friendService={}
	var BASE_URL="http://localhost:8084/RestController"
		
		friendService.listOfSuggestedUsers=function(){
		return $http.get(BASE_URL + "/getsuggestedusers")
	   }
	
	friendService.sendFriendRequest=function(toId){
		return $http.get(BASE_URL + "/friendrequest/" + toId)
	   }
	
	friendService.pendingRequests=function(){
		return $http.get(BASE_URL + "/pendingrequests")
	   }
	friendService.updatePendingRequest=function(request){  //request is Friend obj (id, fromId, toId, status('A'/'D'))
		//request.id=422  request.fromId=john   request.toId=admin   request.status=A
		return $http.put(BASE_URL + "/updatependingrequest", request)   //request obj with updated status
	   }
	
	friendService.listOfFriends=function(){
		return $http.get(BASE_URL + "/friendslist")
	   }
	
	return friendService;
	})

File: BackEndC/src/main/java/com/collaborate/Service/FriendService.java
package com.collaborate.Service;

import java.util.List;

import com.collaborate.Model.Friend;
import com.collaborate.Model.User;

public interface FriendService {
	
List<User> listOfSuggestedUsers(String username);
void friendRequest(Friend friend);

List<Friend> pendingRequests(String toId);

void updateFriendRequest(Friend friend);

List<String> listOfFriends(String username);

}
